cpp/ql/src/Security/CWE/CWE-416/UseOfStringAfterLifetimeEnds.qhelp

```diff
@@ -8,6 +8,12 @@
 When the <code>std::string</code> object is destroyed, the pointer returned by <code>c_str</code> is no
 longer valid. If the pointer is used after the <code>std::string</code> object is destroyed, then the behavior is undefined.
 </p>
+
+<p>Typically this problem occurs when a <code>std::string</code> is returned by a function call (or overloaded operator)
+by value, and the result is not immediately stored in a variable by value (or <code>const</code> reference). The resulting
```

Comment:
There's a third option as well (which is probably the most idiomatic way to fix this nowadays): Storing the temporary string in a rvalue reference. For example:
```cpp
std::string&& s = std::string("abc");
```
The benefit of this (as opposed to storing it in a `const std::string& s`) is that the variable remains mutable.

Maybe it's better to say something like:
> Typically this problem occurs when a <code>std::string</code> is returned by a function call (or overloaded operator)
by value, and the result is not immediately stored in a variable in a way that extends the lifetime of the temporary object

and then we can maybe add a link to https://en.cppreference.com/w/cpp/language/reference_initialization#Lifetime_of_a_temporary?

Reply:
Looks good. I'm happy with this!